src/System.Net.Utilities/src/System/Net/NetworkInformation/Ping.Unix.cs

```diff
@@ -2,33 +2,268 @@
+
+                // We have exceeded our tiemout duration, and no reply has been received.
+                sw.Stop();
+                throw new PingException(SR.net_ping_reply_timeout);
+            }
+        }
+
+        private PingReply SendWithPingUtility(IPAddress address, byte[] buffer, int timeout, PingOptions options)
+        {
+            bool isIpv4 = address.AddressFamily == AddressFamily.InterNetwork;
+            string pingExecutable = isIpv4 ? s_discoveredPing4UtilityPath : s_discoveredPing6UtilityPath;
+            if (pingExecutable == null)
+            {
+                throw new PlatformNotSupportedException(SR.net_ping_utility_not_found);
+            }
+
+            StringBuilder sb = new StringBuilder();
+            sb.Append("-c 1"); // Just send a single ping ("count = 1")
+
+            // The command-line flags for "Do-not-fragment" and "TTL" are not standard.
+            // In fact, they are different even between ping and ping6 on the same machine.
+
+            // The ping utility is not flexible enough to specify an exact payload.
+            // But we can at least send the right number of bytes.
+            sb.Append(" -s ");
+            sb.Append(buffer.Length);
+
+            sb.Append(" ");
+            sb.Append(address.ToString());
+
+            string processArgs = sb.ToString();
+            ProcessStartInfo psi = new ProcessStartInfo(pingExecutable, processArgs);
+            psi.RedirectStandardOutput = true;
+            psi.RedirectStandardError = true;
+            Process p = new Process() { StartInfo = psi };
+            p.Start();
+            if (!p.WaitForExit(timeout))
+            {
+                throw new PingException(SR.net_ping_reply_timeout);
+            }
+
+            if (p.ExitCode != 0)
+            {
+                // This means no reply was received, although transmission may have been successful.
+                throw new PingException(SR.net_ping_reply_timeout);
+            }
+
+            string output = p.StandardOutput.ReadToEnd();
+            int timeIndex = output.IndexOf("time=", StringComparison.Ordinal);
+            int afterTime = timeIndex + 5;
+            double parsedRtt;
+            int msIndex = output.IndexOf("ms", afterTime);
+            int numLength = msIndex - afterTime - 1;
+            string timeSubstring = output.Substring(afterTime, numLength);
+            parsedRtt = double.Parse(timeSubstring);
+            long rtt = (long)Math.Round(parsedRtt);
+            return new PingReply(
+                    address,
+                    null, // Ping utility cannot accomodate these, return null to indicate they were ignored.
+                    IPStatus.Success,
+                    rtt,
+                    Array.Empty<byte>()); // Ping utility doesn't deliver this info.
+        }
+
         private void InternalDisposeCore()
         {
-            // TODO: Implement this (#2487)
+        }
+
+        // Must be 8 bytes total.
```

Comment:
That surprises me, but regardless, you could put the debug-only cctor on the containing Ping class instead.